pb/custom.go

```diff
@@ -0,0 +1,110 @@
+package pstore_pb
+
+import (
+	"encoding/json"
+
+	proto "github.com/gogo/protobuf/proto"
+	peer "github.com/libp2p/go-libp2p-peer"
+	pt "github.com/libp2p/go-libp2p-peer/test"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// customGogoType aggregates the interfaces that custom Gogo types need to implement.
+// it is only used for type assertions.
+type customGogoType interface {
+	proto.Marshaler
+	proto.Unmarshaler
+	json.Marshaler
+	json.Unmarshaler
+	proto.Sizer
+}
+
+// ProtoAddr is a custom type used by gogo to serde raw peer IDs into the peer.ID type, and back.
+type ProtoPeerID struct {
```

Comment:
Figured we might address the TODO while on it: https://github.com/libp2p/go-libp2p-peer/pull/41

Reply:
Off the top of my head, we can't push the serde methods to Multiaddr because the latter is an interface. It works for marshalling but it doesn't for unmarshalling I think.

Reply:
Oh... Hm, I wonder if we should expose the actual multiaddr type? Although we can deal with that later.